app/src/main/java/kth/jjve/xfran/viewmodels/HomeVM.java

```diff
@@ -1,5 +1,155 @@
 package kth.jjve.xfran.viewmodels;
 
-public class HomeVM {
+import android.annotation.SuppressLint;
+import android.app.Application;
+import android.content.Context;
+import android.util.Log;
 
+import androidx.annotation.NonNull;
+import androidx.lifecycle.AndroidViewModel;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+
+import kth.jjve.xfran.models.Workout;
+import kth.jjve.xfran.repositories.LocalStorage;
+import kth.jjve.xfran.repositories.UserRepo;
+import kth.jjve.xfran.repositories.WorkoutRepo;
+
+public class HomeVM extends AndroidViewModel {
+    private MutableLiveData<String> mUserName, mWODTitle, mWODDescription;
+    private MutableLiveData<ArrayList<String>> mWODExercises;
+    private int wodChooser, previousWOD;
+
+    private UserRepo mUserRepo;
+    private WorkoutRepo mWorkoutRepo;
+    private LocalStorage mLocalStorage;
+    @SuppressLint("StaticFieldLeak")
+    private Context mContext;
+
+
+    public HomeVM(@NonNull Application application) {
+        super(application);
+    }
+
+    public void init(Context context) {
+        mContext = context;
+        mUserRepo = UserRepo.getInstance();
+        mUserName = mUserRepo.getUserName();
+
+        mWorkoutRepo = WorkoutRepo.getInstance();
+        try {
+            setWOD(mWorkoutRepo.getWorkouts());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void setWOD(MutableLiveData<List<Workout>> mWorkouts) {
+        // Todo: mDate now gets overwritten when activity is destroyed, find a way to make sure it does not
+        if (checkLastUpdate()) {
+
+            // Get a random workout
+            Random rand = new Random();
+            while (wodChooser == previousWOD) {
+                wodChooser = rand.nextInt(Objects.requireNonNull(mWorkouts.getValue()).size());
+            }
```

Comment:
i think it would make sense that the random wod was not the 2 or 3 previous ones, but not a priority.

Reply:
that is indeed for later